exercises/sam_jenkins/ex02_oo_basics/src/main.cpp

```diff
@@ -0,0 +1,41 @@
+/**
+ * Skeleton main routine
+ */
+#include <string>
+#include <vector>
+#include <iostream>
+#include "Shapes.h"
+#include "Square.h"
+#include "Rectangle.h"
+#include "Circle.h"
+#include "Triangle.h"
+#include "ShapeSorter.h"
+using std::cout; using std::cin; using std::endl; using std::string; using std::vector;
+
+int main(int, char **)
+{
+	vector<Shapes*> shapes(10);
+	shapes[0] = new Square(5);
```

Comment:
Do not use `new` and `delete`. Either use a `unique_ptr` if you only need one reference or a `shared_ptr` if you need reference counting. Only use raw pointers for non-owning references